### PR TITLE
Fix coverity warnings for vulnerability detector

### LINKED ISSUE #21640: Review defects present at Vulnerability Detector
|Wazuh version|Component|Install type|Source|
|---|---|---|--|
| 4.8.0 Alpha 2 | Vulnerability Detector | Manager | #21510|

The following defects reported by Coverity are expected to be fixed by #21294 on branch _4.8.0_. However, they were not marked as solved:

|CID|Name|
|--|--|
|1575836|COPY_INSTEAD_OF_MOVE|
|1575800|COPY_INSTEAD_OF_MOVE|
|1575795|COPY_INSTEAD_OF_MOVE|
|1575794|Use of auto that causes a copy|
|1575776|COPY_INSTEAD_OF_MOVE|
|1575761|COPY_INSTEAD_OF_MOVE|
|1575749|Use of auto that causes a copy|

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #21640 |

## Description

This PR fixes some coverity warnings:
- At `src/data_provider/src/packages/packageLinuxParserHelper.h` the **auto** variable was set as an automatic const reference
- At `src/shared_modules/content_manager/src/actionOrchestrator.hpp` we force the copy with `std::string` because we need to keep the value
- At `src/shared_modules/indexer_connector/src/indexerConnector.cpp`  an **auto** variable was set to reference, and the some parameters were moved to avoid copies. This is considered correct because those variables aren't used in other place.
- At `src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp`, the copy was avoided by creating a member variable and capturing by reference in the lambdas
- At `src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp` the **auto** variable was set as an automatic const reference

## Tests

  - [ ] Coverity

### CODE REVIEW

**Comment by sebasfalcone** [comment]
@pereyra-m Please update the PR body with the new changes

**Review by sebasfalcone — APPROVED** [review]

File: src/shared_modules/indexer_connector/src/indexerConnector.cpp [hunk 1]
```diff
@@ -199,7 +199,10 @@ IndexerConnector::IndexerConnector(
                         sleepTime = std::chrono::seconds(MAX_WAIT_TIME);
                     }
 
-                    initialize(templateData, indexName, selector, secureCommunication);
+                    initialize(std::move(templateData),
+                               std::move(indexName),
+                               std::move(selector),
+                               std::move(secureCommunication));
```
Comment (sebasfalcone) [inline src/shared_modules/indexer_connector/src/indexerConnector.cpp:205]:
```suggestion
                    initialize(templateData, indexName, selector, secureCommunication);
```

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.hpp [hunk 2]
```diff
@@ -59,6 +59,7 @@ class VulnerabilityScannerFacade final : public Singleton<VulnerabilityScannerFa
     void stop();
 
 private:
+    std::shared_ptr<ScanOrchestrator> m_scanOrchestrator;
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.hpp:62]:
```suggestion
```

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp [hunk 3]
```diff
@@ -99,31 +99,31 @@ void VulnerabilityScannerFacade::start(
         policyManager.addSubscriber(m_databaseFeedManager);
 
         // Init Orchestrator
-        auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
+        m_scanOrchestrator = std::make_shared<ScanOrchestrator>(
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp:102]:
```suggestion
        auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
```

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp [hunk 4]
```diff
@@ -99,31 +99,31 @@ void VulnerabilityScannerFacade::start(
...
             m_indexerConnector, m_databaseFeedManager, m_reportDispatcher, m_internalMutex);
 
         // Subscription to syscollector delta events.
         m_syscollectorDeltasSubscription =
             std::make_unique<RouterSubscriber>("deltas-syscollector", "vulnerability_scanner_deltas");
         m_syscollectorDeltasSubscription->subscribe(
-            [scanOrchestrator](const std::vector<char>& message)
+            [&](const std::vector<char>& message)
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp:109]:
```suggestion
            [scanOrchestrator](const std::vector<char>& message)
```

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp [hunk 5]
```diff
@@ -99,31 +99,31 @@ void VulnerabilityScannerFacade::start(
...
             {
                 std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
                     SyscollectorDeltas::GetDelta(message.data());
 
-                scanOrchestrator->run(data);
+                m_scanOrchestrator->run(data);
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp:114]:
```suggestion
                scanOrchestrator->run(data);
```

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp [hunk 6]
```diff
@@ -99,31 +99,31 @@ void VulnerabilityScannerFacade::start(
...
             });
 
         // Subscription to syscollector rsync events.
         m_syscollectorRsyncSubscription =
             std::make_unique<RouterSubscriber>("rsync-syscollector", "vulnerability_scanner_rsync");
         m_syscollectorRsyncSubscription->subscribe(
-            [scanOrchestrator](const std::vector<char>& message)
+            [&](const std::vector<char>& message)
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp:121]:
```suggestion
            [scanOrchestrator](const std::vector<char>& message)
```

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp [hunk 7]
```diff
@@ -99,31 +99,31 @@ void VulnerabilityScannerFacade::start(
...
             {
                 std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
                     SyscollectorSynchronization::GetSyncMsg(message.data());
 
-                scanOrchestrator->run(data);
+                m_scanOrchestrator->run(data);
```
Comment (sebasfalcone) [inline src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp:126]:
```suggestion
                scanOrchestrator->run(data);
```

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**